webpack/assets/javascripts/react_app/routes/FiltersForm/FiltersForm.js

```diff
@@ -0,0 +1,259 @@
+        >
+          <Checkbox
+            isChecked={isOverride}
+            onChange={checked => {
+              setIsOverride(checked);
+            }}
+            aria-label="is override"
+            id="override-check"
+            name="override"
+          />
+        </FormGroup>
+      )}
+      {isOverride && (
+        <Taxonomies
+          showOrgs={showOrgs}
+          showLocations={showLocations}
+          setChosenOrgs={setChosenOrgs}
+          setChosenLocations={setChosenLocations}
+          defaultOrgs={data.organizations}
+          defaultLocations={data.locations}
+        />
+      )}
+      {isGranular ? (
+        <>
+          <FormGroup
+            label={__('Unlimited?')}
+            labelIcon={
+              <Popover
+                bodyContent={
+                  <div>
+                    {__(
+                      'If the unlimited field is enabled, the filter applies to all resources of the selected type. If the unlimited  field is disabled, you can specify further filtering using Foreman search syntax in the search field. If the role is associated with organizations or locations, the filters are not considered unlimited as they are scoped accordingly.'
+                    )}
+                  </div>
+                }
+              >
+                <button
+                  type="button"
+                  aria-label="More info for unlimited field"
+                  onClick={e => e.preventDefault()}
+                  className="pf-c-form__group-label-help"
+                >
+                  <HelpIcon noVerticalAlign />
+                </button>
+              </Popover>
+            }
+          >
+            <Checkbox
+              isChecked={isUnlimited}
+              onChange={checked => {
+                setIsUnlimited(checked);
+                dispatch(updateDisability(checked, SEARCH_ID));
+              }}
+              aria-label="is unlimited"
+              id="unlimited-check"
+              name="unlimited"
+            />
+          </FormGroup>
+          <FormGroup label={__('Search')} className="filter-form-search">
+            <SearchBar
+              disabled={!isUnlimited}
+              initialQuery={data.search}
+              data={{
+                controller: type.name,
+                autocomplete: {
+                  id: SEARCH_ID,
+                  url: type.search_path,
+                },
+              }}
+            />
+          </FormGroup>
+        </>
+      ) : (
+        <Alert
+          variant="info"
+          title={__(
+            "Selected resource type does not support granular filtering, therefore you can't configure granularity"
+          )}
+        />
+      )}
+
+      <ActionGroup>
+        <Button
+          variant="primary"
+          isDisabled={!chosenPermissions.length}
+          onClick={submit}
+        >
+          {__('Submit')}
+        </Button>
+        <Button variant="link">{__('Cancel')}</Button>
```

Comment:
clicking cancel should bring the user back to the role page, isn't it?

Reply:
It should, added an `onClick`